lib/Feature/FeatureModelParser.cpp

```diff
@@ -286,10 +275,364 @@ std::unique_ptr<xmlDoc, void (*)(xmlDocPtr)> FeatureModelXmlParser::parseDoc() {
   } else {
     llvm::errs() << "Failed to parse / validate XML.\n";
   }
-  return std::unique_ptr<xmlDoc, void (*)(xmlDocPtr)>(nullptr, nullptr);
+  return UniqueXmlDoc(nullptr, nullptr);
 }
 
 // TODO(s9latimm): replace with builder err
 bool FeatureModelXmlParser::verifyFeatureModel() { return parseDoc().get(); }
 
+//===----------------------------------------------------------------------===//
+//                        FeatureModelSxfmParser Class
+//===----------------------------------------------------------------------===//
+
+std::unique_ptr<FeatureModel> FeatureModelSxfmParser::buildFeatureModel() {
+  UniqueXmlDoc Doc = parseDoc();
+  if (!Doc) {
+    return nullptr;
```

Comment:
Why do you return a `nullptr` here?
maybe throw an error or call a method that terminates the parser?
Or is this expected valid behaviour, that the Doc was empty?

Reply:
`buildFeatureModel` is actually the entry point for the parser, so returning here terminates the parser as well.

I returned `nullptr` for the XML part at this point too, as it is easy to test afterwards if parsing succeeded and somewhat recoverable, but I think an error message would not harm.

On the other hand, an empty input will always be rejected as it would be invalidated through the DTD.

Reply:
1. The behavior is consistent to what the XML parser does. 
2. IF line 291 is reached, then errors are already printed out on the console. Error handling is done in the parseDoc() method that is called in line 289. I do not think that we need further error handling. 

But I am free for suggestions :blush: